src/hotspot/cpu/x86/assembler_x86.cpp

```diff
@@ -14534,10 +15864,26 @@ void Assembler::sarq(Register dst, int imm8) {
   }
 }
 
+void Assembler::esarq(Register dst, Register src, int imm8, bool no_flags) {
+  InstructionAttr attributes(AVX_128bit, /* vex_w */ true, /* legacy_mode */ false, /* no_mask_reg */ true, /* uses_vl */ false);
+  int encode = evex_prefix_and_encode_ndd(0, dst->encoding(), src->encoding(), VEX_SIMD_NONE, VEX_OPCODE_0F_3C, &attributes, no_flags);
+  if (imm8 == 1) {
+    emit_int16((unsigned char)0xD1, (0xF8 | encode));
+  } else {
+    emit_int24((unsigned char)0xC1, (0xF8 | encode), imm8);
+  }
+}
+
 void Assembler::sarq(Register dst) {
   int encode = prefixq_and_encode(dst->encoding());
   emit_int16((unsigned char)0xD3, (0xF8 | encode));
 }
+
+void Assembler::esarq(Register dst, Register src, bool no_flags) {
+  InstructionAttr attributes(AVX_128bit, /* vex_w */ true, /* legacy_mode */ false, /* no_mask_reg */ true, /* uses_vl */ false);
+  int encode = evex_prefix_and_encode_ndd(0, dst->encoding(), src->encoding(), VEX_SIMD_NONE, VEX_OPCODE_0F_3C, &attributes, no_flags);
+  emit_int16((unsigned char)0xD1, (0xF8 | encode));
```

Comment:
This should be:
 emit_int16((unsigned char)0xD3, (0xF8 | encode));
Shift by cl and not shift by 1.

Reply:
Thanks, fixed.